internal/ethapi/api.go

```diff
@@ -1889,6 +1889,36 @@ func (api *PrivateDebugAPI) SetHead(number hexutil.Uint64) {
 	api.b.SetHead(uint64(number))
 }
 
+// BurnedETH returns the amount of burned ether for a specific
+// range of blocks under eip-1559.
+// The returned value is in denoted in Wei.
+func (api *PrivateDebugAPI) BurnedETH(start, end *hexutil.Uint64) (*hexutil.Big, error) {
+	burned := big.NewInt(0)
+	startBlock := 0
+	endBlock := int(api.b.CurrentHeader().Number.Int64())
+	if start != nil {
+		startBlock = int(*start)
+	}
+	if end != nil {
+		endBlock = int(*end)
+	}
+	if startBlock > endBlock {
+		return (*hexutil.Big)(burned), errors.New("invalid range specified, start > end")
+	}
+	for i := startBlock; i < endBlock; i++ {
+		block, err := api.b.BlockByNumber(context.Background(), rpc.BlockNumber(i))
+		if err != nil {
+			return (*hexutil.Big)(burned), err
+		}
+		txs := len(block.Transactions())
+		if basefee := block.Header().BaseFee; basefee != nil && basefee.Cmp(common.Big0) != 0 {
+			burnedBlock := new(big.Int).Mul(basefee, big.NewInt(int64(txs)))
```

Comment:
You're calculating the burn as in `baseFeePerTx`, should be `baseFeePerGas`, so shouldn't you instead take the amount of gas used in the block and multiply with the basefee?

Reply:
Yes. You also don't need to entire block for it, you can use `header.gasUsed * header*baseFee`

Reply:
Yeag, I'm dumb^^ will change, of course basefee is per gas not per transaction